### PR TITLE
Fix change_map command not always changing (close #1841)

### LINKED ISSUE #1841: change_map command does not work
Hey,

today I tried to change the map on my server with change_map ctf2 in rcon and it did not work. No reaction by the server. Only sv_map worked. 

Did I do anything wrong? Whats actually the difference between sv_map and change_map ?

Cheers!

### PR DESCRIPTION
**Pull request by sepulzera**
After applying this PR, the change_map command will work even
- when in warmup
- when the map is not finished (sv_matches_per_map)

As this is my first PR for teeworlds, I hope to satisfy the code guidelines and quality standard :-)

### CODE REVIEW

**Comment by fokkonaut** [comment]
just remove that command, we have sv_map.

**Comment by Dune-jr** [comment]
Seems good.
I wouldn't mind `change_map` to no longer work with a parameter though.

File: src/game/server/gamecontroller.cpp [hunk 1]
```diff
@@ -747,7 +747,14 @@ void IGameController::Tick()
 				break;
 			case IGS_END_MATCH:
 				// start next match
-				CycleMap();
+				if(m_MatchCount >= m_GameInfo.m_MatchNum-1)
+				{
+					CycleMap();
+					break;
```
Comment (oy) [inline src/game/server/gamecontroller.cpp]: this breaks the game in case sv_maprotation is empty.

Comment (sepulzera) [reply src/game/server/gamecontroller.cpp]: Thanks for the nice catch, oy!
I removed the break, now it works as intended, with and without sv_maprotation being set.

**Pull request merged**